crypto/dh/dh_kdf.c

```diff
@@ -185,3 +190,4 @@ int DH_KDF_X9_42(unsigned char *out, size_t outlen,
     EVP_MD_CTX_cleanup(&mctx);
     return rv;
 }
+#endif
```

Comment:
@levitte I've tried clang in the meantime as well.
I'm using gcc-4.9 and clang-3.9.1. I've added -DPEDANTIC to both debug-linux-x86_64 and debug-linux-x86_64-clang and neither one fails:
    make clean
    ./Configure no-cms no-dh no-ec 
                    debug-linux-x86_64-clang
                    debug-linux-x86_64
// these targets are not modified and pass as expected:
                    linux-x86_64-clang
                    linux-x86_64
    (make depend)
    make

I don't have a clear understanding of the issue fixed by the dummy line and some of the commit messages I've looked into did not help. OpenSSL 1.1.0 from where I've cherry-picked this commit does not have the dummy line either.

I can easily add the #else branch but I don't feel comfortable doing this without some basic checks. Do you have any pointers that I can try?